DOC: Extend explanation on single vs double backticks. (#565)
numpy/numpydoc: doc/format.rst

whereas the behavior of single backticks is defined by the default role. This
leads to the following style recommendations:

> - Any object that can be linked to should be enclosed in single backticks. That
>   is to say : Module, class, function, method, and attribute names are some of
>   the objects that should be enclosed in single backticks. On a case by cases,

Reviewer: I think "for example" is more appropriate than "that is" here because it gives examples rather than restating. Also, I assume you meant "on a case-by-case basis"
```suggestion
- For example, module, class, function, method, and attribute names
  should be enclosed in single backticks. On a case-by-case basis,
```

Maybe we do legitimately disagree a little bit. I do think that text that refers to code should always be monospaced. For objects that happen to link, the way to accomplish this is to enclose them in single backticks.

Reviewer: Oops, I think I messed up here. I didn't mean to remove the first part: "Any object that can be linked to should be enclosed in single backticks." Otherwise, the change looks like what I had in mind.

  instances may also be included in single backticks, like :py:obj:`None`,
  :py:obj:`True`, :py:obj:`False`.

  These will typically render as hyperlinks in monospaced font (e.g.
  :any:`numpy`) to the relevant object documentation; depending on project
  settings. If the hyperlink does not render properly or point to the intended
  object, explicitly include the appropriate role and/or namespace.

- This guide continues to recommended that parameter names be enclosed within
  single backticks. Currently, this may cause parameter names to render
  improperly and cause warnings, but numpydoc will soon release a feature
  that causes them to render as monospaced hyperlinks to the parameter
  documentation.

- All other text that is intended to render in ``monospaced`` font and is not

>   supposed to refer to another docs page should be enclosed within ````double

Reviewer: I thought "and is not supposed to refer to another docs page" was implied because this says "All *other* text" and it appears after we've discussed links. Does it need to be more explicit?

Author: I think we can also be more explicit, and you can have monospaced linked to another page if you use a directive. like `` :ref:`the text <the_ref>` `` technically I think. I'm just also trying to separate rendering from semantic.

Author: But I'm happy to remove.

Reviewer: I read it as redundant, but if others don't, it's fine!

whereas the behavior of single backticks is defined by the default role. This
leads to the following style recommendations:

- For example, module, class, function, method, and attribute names
  should be enclosed in single backticks. On a case-by-case basis,
  instances may also be included in single backticks, like :py:obj:`None`,

>   :py:obj:`True`, :py:obj:`False`.

Reviewer: @Carreau this renders like:
<img width="925" alt="image" src="https://github.com/user-attachments/assets/59097496-130c-4885-919a-69f15d7c668e" />

Seeing only the rendered version and not the underlying reST, I would assume from the text that these object names only need to be included in single backticks to render like this, but that's not quite right. Should these go at the end of the paragraph, after the idea of explicitly including the role and/or namespace is mentioned?

  These will typically render as hyperlinks in monospaced font (e.g.

>   :any:`numpy`) to the relevant object documentation; depending on project
>   settings. If the hyperlink does not render properly or point to the intended

Reviewer: ```suggestion
  :any:`numpy`) to the relevant object documentation.
  If the hyperlink does not render properly or point to the intended
```
Semicolon is no longer the appropriate punctuation, and I think this part can just be removed now.
